src/xcode/ENA/ENA/Source/Models/SymptomsOnset.swift

```diff
@@ -0,0 +1,91 @@
+//
+// Corona-Warn-App
+//
+// SAP SE and all other contributors
+// copyright owners license this file to you under the Apache
+// License, Version 2.0 (the "License"); you may not use this
+// file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+import Foundation
+import ExposureNotification
+
+enum SymptomsOnset {
+
+	case noInformation
+	case nonSymptomatic
+	case symptomaticWithUnknownOnset
+	case lastSevenDays
+	case oneToTwoWeeksAgo
+	case moreThanTwoWeeksAgo
+	case daysSinceOnset(Int)
+
+	// MARK: - Internal
+
+	/// Transmission risk level by days since the exposure.
+	/// These factors are applied to each `ENTemporaryExposureKey`'s `transmissionRiskLevel`
+	///
+	/// Index 0 of the array represents the day of the exposure
+	/// Index 1 the next day, and so on.
+	/// These factors are supplied by RKI
+	///
+	/// - see also: [Risk Score Calculation Docs](https://github.com/corona-warn-app/cwa-documentation/blob/master/solution_architecture.md#risk-score-calculation)
+	var transmissionRiskVector: [ENRiskLevel] {
+		switch self {
+		case .noInformation:
+			return [5, 6, 7, 7, 7, 6, 4, 3, 2, 1, 1, 1, 1, 1, 1]
+		case .nonSymptomatic:
+			return [4, 4, 3, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]
+		case .symptomaticWithUnknownOnset:
+			return [5, 6, 8, 8, 8, 7, 5, 3, 2, 1, 1, 1, 1, 1, 1]
+		case .lastSevenDays:
+			return [4, 5, 6, 7, 7, 7, 6, 5, 4, 3, 2, 1, 1, 1, 1]
+		case .oneToTwoWeeksAgo:
+			return [1, 1, 1, 1, 2, 3, 4, 5, 6, 6, 7, 7, 6, 6, 4]
+		case .moreThanTwoWeeksAgo:
+			return [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 3, 4, 5]
+		case .daysSinceOnset(let daysSinceOnset):
+			precondition(daysSinceOnset < 22)
```

Comment:
this precondition would make the app terminate in a release build if it is violated, would assert make more sense here? you already handle the issue by using min(daysSinceOnset, 21) in the next line (great! 🥇). Or do you prefer to explicitly fail here?

Reply:
Oh, you're right, I wanted it to continue running in production, must have mixed up `precondition` and `assert`. Have changed it to assert now ☺️